fdbcli/fdbcli.actor.cpp

```diff
@@ -2553,6 +2553,27 @@ Future<T> stopNetworkAfter( Future<T> what ) {
 	}
 }
 
+ACTOR Future<Void> addInterface( std::map<Key,std::pair<Value,ClientLeaderRegInterface>>* address_interface, Reference<FlowLock> connectLock, KeyValue kv) {
+	wait(connectLock->take());
+	state FlowLock::Releaser releaser(*connectLock);
+	state ClientWorkerInterface workerInterf = BinaryReader::fromStringRef<ClientWorkerInterface>(kv.value, IncludeVersion());
+	state ClientLeaderRegInterface leaderInterf(workerInterf.address());
+	choose {
+		when( Optional<LeaderInfo> rep = wait( brokenPromiseToNever(leaderInterf.getLeader.getReply(GetLeaderRequest())) ) ) {
+			StringRef ip_port = kv.key.endsWith(LiteralStringRef(":tls")) ? kv.key.removeSuffix(LiteralStringRef(":tls")) : kv.key;
+			(*address_interface)[ip_port] = std::make_pair(kv.value, leaderInterf);
+
+			if(workerInterf.reboot.getEndpoint().addresses.secondaryAddress.present()) {
+				Key full_ip_port2 = StringRef(workerInterf.reboot.getEndpoint().addresses.secondaryAddress.get().toString());
+				StringRef ip_port2 = full_ip_port2.endsWith(LiteralStringRef(":tls")) ? full_ip_port2.removeSuffix(LiteralStringRef(":tls")) : full_ip_port2;
+				(*address_interface)[ip_port2] = std::make_pair(kv.value, leaderInterf);
+			}
```

Comment:
This bit is new, is it necessary to issue kills to both interfaces? Is this going to print duplicates in our list?

Reply:
It shouldn't be necessary to issue kills to both interfaces, but we want to allow people to issue kills to either interface. I've found that it's hard to predict which interface is going to appear in the kill list today.

Reply:
Oh, that's a good point.

Reply:
If we happen to issue kills to the same process twice because of this, is it possible that the process gets killed twice, or is there something that protects us from that?

Reply:
Once a worker is killed, it will come back with a different interface, so it is safe to send multiple kills.